internal/server/proto/server.proto

```diff
@@ -2399,6 +2410,18 @@ message StatusReport {
   // id is the unique ID for this status report.
   string id = 6;
 
+  // data_source_poll starts at number 30 to reserve any space for copied status
+  // report options on this message. Hopefully with RFC WP-072, any further
+  // status report details needed out of the raw report can be obtained through
+  // the json structure rather than manually copied and added here.
+
+  // Each polling event is tracked as a separate job. You can query the
+  // poll operations and their success/failure by using the ListJobs API.
+  Poll data_source_poll = 30;
```

Comment:
This feels a bit weird. Why is the polling settings _per_ status report? Shouldn't the polling settings for status reports be on the application or project? That way you're saying "For this project/app I want a status report every X seconds" ?

Reply:
Hmm, right, that makes sense. In that case, I'd probably place polling on the application, and I don't think this PR is really needed then for polling status reports per app. I'm pretty sure then I can essentially use the built-in project polling state funcs since application is built off project. So just adding an ApplicationPollPeek/ApplicationPollComplete similar to how its other funcs already use project.

App Operation polling still might come in handy in the future so it's probably still useful to go forward with this minus the status report poll message changes 🤔 If not I could close this and redo it based on the plan ☝🏻

Reply:
Yeah app operation polling might still come in handy one day. Agreed you probably don't need the status report stuff. 

And yeah I'd create an implementation of `applicationPollHandler` similar to project poll handler. I don't think we need to expose any user facing settings, just run each application that has a deployment/release like every 30 seconds or something (we can tune that later). 

Might be worth talking to Evan to hear his thoughts while I'm out.